<a href="https://colab.research.google.com/github/Nzaya/Detecting-Parkinson-s-Disease/blob/master/Detecting_Parkinson%E2%80%99s_Disease.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Installing necessary libraries 

import pandas as pd
import numpy as np
import os, sys
from sklearn.preprocessing import MinMaxScaler
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

In [2]:
# Load the dataset

df = pd.read_csv('/content/parkinsons.csv')
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
0  phon_R01_S01_1      119.992       157.302  ...  0.266482  2.301442  0.284654
1  phon_R01_S01_2      122.400       148.650  ...  0.335590  2.486855  0.368674
2  phon_R01_S01_3      116.682       131.111  ...  0.311173  2.342259  0.332634
3  phon_R01_S01_4      116.676       137.871  ...  0.334147  2.405554  0.368975
4  phon_R01_S01_5      116.014       141.781  ...  0.234513  2.332180  0.410335

[5 rows x 24 columns]

In [14]:
df.tail()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
190  phon_R01_S50_2      174.188       230.978  ...  0.121952  2.657476  0.133050
191  phon_R01_S50_3      209.516       253.017  ...  0.129303  2.784312  0.168895
192  phon_R01_S50_4      174.688       240.005  ...  0.158453  2.679772  0.131728
193  phon_R01_S50_5      198.764       396.961  ...  0.207454  2.138608  0.123306
194  phon_R01_S50_6      214.289       260.277  ...  0.190667  2.555477  0.148569

[5 rows x 24 columns]

In [6]:
df['name'].unique

<bound method Series.unique of 0      phon_R01_S01_1
1      phon_R01_S01_2
2      phon_R01_S01_3
3      phon_R01_S01_4
4      phon_R01_S01_5
            ...      
190    phon_R01_S50_2
191    phon_R01_S50_3
192    phon_R01_S50_4
193    phon_R01_S50_5
194    phon_R01_S50_6
Name: name, Length: 195, dtype: object>

In [15]:
# Drop the name column

df.drop(['name'], axis = 1, inplace = True)

In [17]:
# Split our dataset into X and y 
# X are our features and y our labels

X = df.drop('status',1).values
y = df['status'].values

In [18]:
# Count the number of 0 and 1 in the status column

print(y[y==1].shape[0], y[y==0].shape[0])

147 48


We have 147 ones and 48 zeros

# MIN MAX SCALER

Lets scale the features(X) to between -1 and 1 to normalize them, then use the fit_transform method to fit the data then transform it

In [19]:
scaler = MinMaxScaler((-1,1))
x= scaler.fit_transform(X)
y = y

In [20]:
# Split the dataset into training and testing

X_train,X_test,y_train,y_test = train_test_split(x,y,test_size = 0.2, random_state = 7)

In [21]:
# Train the model

model = XGBClassifier()
model.fit(X_train,y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

In [23]:
# Calculate the accuracy

y_pred = model.predict(X_test)
print(accuracy_score(y_test,y_pred)*100)

94.87179487179486
